frontend/src/context/auth-context.tsx

```diff
@@ -1,16 +1,20 @@
 import { createContext, useCallback, useContext, useState } from 'react'
 import { useNavigate } from 'react-router-dom'
+import { AUTH_TOKEN_STORAGE_KEY } from '../constants/session-storage'
 
 type AuthContextType = {
   token: string | null
   logIn: (userName: string, password: string) => void
   logOut: () => void
+  authenticate: (token: string) => void
 }
 
 const AuthContext = createContext(null as AuthContextType | null)
 
 export const AuthProvider = ({ children }: { children: JSX.Element }) => {
-  const [token, setToken] = useState(null as string | null)
+  const [token, setToken] = useState(
+    sessionStorage.getItem(AUTH_TOKEN_STORAGE_KEY) as string | null
```

Comment:
`sessionStorage.getItem` return type is `string | null`, so there is no need for typecasting here.

Reply:
Removed redundant typecast